underscore.js

```diff
@@ -1370,12 +1381,27 @@
 
   _.noop = function(){};
 
-  _.property = property;
+  _.property = function(path) {
+    if (!_.isArray(path)) {
+      return shallowProperty(path);
+    }
+    var length = path.length;
+    return function(obj) {
+      for (var i = 0; i < length; i++) {
+        if (obj == null) return void 0;
+        obj = obj[path[i]];
+      }
+      return length ? obj : void 0;
+    };
+  };
 
   // Generates a function for a given object that returns a given property.
   _.propertyOf = function(obj) {
-    return obj == null ? function(){} : function(key) {
-      return obj[key];
+    if (obj == null) {
+      return function(){};
+    }
+    return function(key) {
+      return _.property(key)(obj);
```

Comment:
My thought was that eventually the deep case for both `_.property` and `propertyOf` could call out to `_.get`. Or we could give each their own implementation. Do you think the weight savings is worth the perf hit? I was feeling self-conscious about my weight ;)